ldaptor/protocols/ldap/ldapclient.py

```diff
@@ -204,9 +206,11 @@ def handle(self, msg):
                 # Return true to mark request as fully handled
                 if return_controls:
                     if handler(msg.value, msg.controls, *args, **kwargs):
+                        d.callback(None)
                         del self.onwire[msg.id]
                 else:
                     if handler(msg.value, *args, **kwargs):
+                        d.callback(None)
```

Comment:
The functional changes are these two `d.callback` lines. This lets callers of `send_multiResponse` and `send_multiResponse_ex` yield on those methods if they want to yield until all responses have been handled.